enhancements/autoscaling/provisioning-request-enablement.md

```diff
@@ -0,0 +1,337 @@
+  flag when deploying the Cluster Autoscaler.
+* Update the Cluster Autoscaler RBAC to allow modification of ProvisioningRequests.
+
+#### Standalone Clusters
+
+Standalone clusters utilize the Cluster Autoscaler Operator to deploy and operate the Cluster
+Autoscaler. This operator currently installs resource definitions for ClusterAutoscaler and
+MachineAutoscaler resource types. This enhancements proposes the following changes:
+
+* The Cluster Autoscaler Operator will install the `provisioningrequests.autoscaling.x-k8s.io`
+  resource definition in its installation manifests.
+* The Cluster Autoscaler Operator will always enable the `--enable-provisioing-requests`
+  flag when deploying the Cluster Autoscaler.
+* Update the Cluster Autoscaler RBAC to allow modification of ProvisioningRequests.
+
+#### Single-node Deployments or MicroShift
+
+This change does not affect single-node deployments and Microshift as they do not
+install the Cluster Autoscaler Operator.
+
+### Implementation Details/Notes/Constraints
+
+The main implementation detail for this enhancement is the inclusion of the
+ProvisioningRequest custom resource definition.
+
+Currently, the Cluster Autoscaler Operator contains custom resource definitions for
+the ClusterAutoscaler and MachineAutoscaler types in its payload manifests. Following
+in this pattern, the `Makefile` for the Cluster Autoscaler Operator will be modified to
+include a target for rendering the ProvisioningRequest manifest from vendored files
+containing the Cluster Autoscaler API package. The resulting custom resource definition
+manifest will be included in the payload manifests for the operator.
+
+The second implementation detail worth noting is the change in default configuration
+for the Cluster Autoscaler. In order to properly reconcile ProvisioningRequests in
+a cluster, the Cluster Autoscaler must be started with the command line flag
+`--enable-provisioning-requests=true`. The Cluster Autoscaler Operator will be
+updated to include this flag as a default. There will not be a corresponding
+API field to allow user configuration of this flag.
+
+### Risks and Mitigations
+
+This change is generally low risk but there are a few concerns to consider:
+
+1. Unauthorized creation of infrastructure resources. Because the ProvisioningRequest
+  resource will lead to infrastructure resource creation, it must have a proper role and
+  permissions for users to create. This will follow the same role-based authentication
+  control pattern as other Kubernetes resources, but is worth noting as creation of
+  ProvisioningRequests could lead to the creation of infrastructure resources.
+2. Confusion for users around the use of ProvisioningRequests. The Cluster Autoscaler will
+  only react to ProvisioningRequests that are created in the `openshift-machine-api`
+  namespace. This will be clearly illustrated with examples in the OpenShift product
+  documentation.
```

Comment:
Why do we have this limitation?

Reply:
i don't think we actually do have this limitation. it appears that the cluster-autoscaler wants to react to ProvisioningRequests in any namespace. i'm not sure if we want this, but that's how it works today.

Reply:
removed this language